From 140b6ec47bbc103febf45ce0c51d1cd4ef50eb06 Mon Sep 17 00:00:00 2001
From: Jason Gerecke <killertofu@gmail.com>
Date: Thu, 9 Feb 2012 11:24:11 -0800
Subject: [PATCH] Check for appropriate number of properties when setting
 buttons

Commit 7e92aca6 changed the length of the Wacom Button Actions
property to be dynamically calculated, but did not take this
into account for the property setter (preventing xsetwacom from
updating them). This patch replaces the constant length with an
identical calculation to ensure we check for the proper size.

Signed-off-by: Jason Gerecke <killertofu@gmail.com>
Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 src/wcmXCommand.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/src/wcmXCommand.c b/src/wcmXCommand.c
index 210ea5d..78a38a3 100644
--- a/src/wcmXCommand.c
+++ b/src/wcmXCommand.c
@@ -847,7 +847,8 @@ int wcmSetProperty(DeviceIntPtr dev, Atom property, XIPropertyValuePtr prop,
 #endif
 	} else if (property == prop_btnactions)
 	{
-		if (prop->size != WCM_MAX_BUTTONS)
+		int nbuttons = min(max(priv->nbuttons + 4, 7), WCM_MAX_BUTTONS);
+		if (prop->size != nbuttons)
 			return BadMatch;
 		wcmSetPropertyButtonActions(dev, property, prop, checkonly);
 	} else
-- 
1.7.7.5

